### PR TITLE
(Feature) (NodeUI) Undo/redo Buttons workflow editor

### PR DESCRIPTION
**Pull request by mickr777**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [x] Yes
- [ ] No, because:

      
## Have you updated all relevant documentation?
- [ ] Yes
- [ ] No

## Description
Cleaned up code based on @psychedelicious suggestions
Ready For Review 

Added code to clear undo/redo history on browser reload
(I felt it need some way to clear it for now)

## Related Tickets & Documents
https://github.com/invoke-ai/InvokeAI/issues/4137

- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings
image

## Added/updated tests?

- [ ] Yes
- [ ] No : _please replace this line with details on why tests
      have not been included_

### CODE REVIEW

**Review by psychedelicious — CHANGES_REQUESTED** [review]
I've requested some changes, and implemented them all locally, but there are some deeper issues:
- Many things we need to have in undo history do not trigger new history entries. For example, node labels, adding nodes to workflow linear UI, node positions
- Some things change rapidly and we need logic to not flood the undo history, eg the position of nodes, node notes/labels
- Node selection state needs to be updated after every history change

I've started addressing these but running into a few weird behaviours that I'll need to spend some time investigating.

Thanks for getting this effort started. I'm happy to take it from here. I'll update this PR when I work on it next.

**Comment by psychedelicious** [comment]
Converted to draft pending resolution of the other issues

**Comment by mickr777** [comment]
> Converted to draft pending resolution of the other issues

Ok, I did try what I could, but I will leave it in your hands now 😃

File: invokeai/frontend/web/src/features/nodes/store/nodesPersistDenylist.ts [hunk 1]
```diff
@@ -15,4 +15,16 @@ export const nodesPersistDenylist: (keyof NodesState)[] = [
...
+  'mouseOverNode',
+  'isAddNodePopoverOpen',
+  'viewport',
+  'shouldShowFieldTypeLegend',
+  'shouldShowMinimapPanel',
+  'shouldValidateGraph',
+  'shouldAnimateEdges',
+  'nodeOpacity',
+  'shouldSnapToGrid',
+  'shouldColorEdges',
```
[the remark this answers is not in the document]
Comment (mickr777) [reply invokeai/frontend/web/src/features/nodes/store/nodesPersistDenylist.ts:29]: Just notice This is the wrong nodesPersistDenylist.ts it must of snuck in from a test I was doing

[the remark this answers is not in the document]
Comment (mickr777) [reply invokeai/frontend/web/src/features/nodes/store/nodesPersistDenylist.ts:29]: Just notice This is the wrong nodesPersistDenylist.ts it must of snuck in from a test I was doing

**Pull request closed without merging**